### DISCUSSION TITLE
.net 6 support

### DISCUSSION THREAD

**Discussion by haesta1066** [dotnet/aspnet-api-versioning#782] [General]
Hi. Are there plans to upgrade to .net 6? Thanks.

**Comment by commonsensesoftware** [comment]
.NET 6 will be supported, but there isn't a specific plan per se. Nothing new beyond bug fix is currently on the radar for that release. That _should_ make the transition smooth. To be honest, I've been at my limit of extra time the past many months. I'm close to being able to fix or resolve the majority of the open issues. OData is the furthest behind. People have been clamoring for 8.0 support, but it requires a rewrite (again) because OData changed everything (again).

If I were to layout a plan, it would effectively go something:

1. Fix non-OData open issues
2. Upgrade  non-OData to .NET 6.0
3. Fix all OData open issues
4. Upgrade OData to 8.0
5. Upgrade OData to 8.0 with .NET 6.0
6. Add new features

In general, things have been pretty stable. The new _Minimal APIs_ certainly through a wrench in the machine. Some significant refactoring will be required to make that work, but I definitely believe it's possible (e.g. I've been thinking about it). Removing the dependency on any part of MVC is _interesting_ and likely appealing in some scenarios.

**Reply by ransht** [reply]
is there any progress ?
its the only package that stops up from moving to NET6

**Comment by xavierjohn** [comment]
I am on .net 6 and using API versioning without any problem.
I am not using OData.

**Comment by surajkumar8642** [comment]
Have any work for me ?
I work on C#, MVC, Entity Framework, SQL

**Comment by commonsensesoftware** [comment]
If you're following this thread, there is a lot of back story that is too much to rehash here. Nevertheless, I have been actively and feverously working on making all this happen. There are big changes coming, but I think we are just about there - or at least the first iteration. My goal is to have a detailed announcement and roadmap by the end of the week. Stay tuned.

**Comment by commonsensesoftware** [comment]
@surajkumar8642 Thank you for your offer. Not a lot of people have offered their time, so I appreciate you. 😉 At this very moment, I don't have a particular _Up For Grabs_ queue. There's a fairly large bug queue that I have mostly burned down, but not merged. There's a huge set of changes that will impact .NET 6+ going forward that I've almost baselined. There will some dev opportunities after that. In particular, I'm considering expanding the wiki to a fully-fledged documentation site, more examples, and templates/scaffolding. Once I reach that point, I'll reach out to see if you're still available and interested. Thanks again.

**Comment by commonsensesoftware** [comment]
Just a quick update. There are some important announcements that have just been made. We're still a little bit off from publishing new packages, but the new [main](../tree/main) branch has .NET 6.0. I hope to have new a preview version of the packages released soon (e.g. ASAP). Thank you for your patience.

**Comment by andrei-micuda** [comment]
Hello! Is the `Asp.Versioning.ApiExplorer`  package fully supported in .NET 6? I am trying to migrate an existing .NET 5 application to .NET 6 and have done the following:

Replaced
```c#
services.AddVersionedApiExplorer( options => { /* configure options */ } );
```
with
```c#
services.AddApiVersioning(options => { /* configure options */ } );
```

This seems to be working fine for the most part with options such as `AssumeDefaultVersionWhenUnspecified` still being present. But other options that exist in .NET 5 seem to be missing from .NET 6. I am mainly talking about `GroupNameFormat` and `SubstituteApiVersionInUrl`.

Am I misusing the service extension method?

Thanks.

**Reply by commonsensesoftware** [reply]
.NET 6 is fully supported, but a number of things have changed, which I hope you find for the better. The wiki is still out-of-date, but all of the code examples are current. That is always a good place to look if you get stuck.

The combination of opt-in services for API Versioning now chain together:

```c#
services.AddApiVersioning()     // Core API Versioning services with support for Minimal APIs
        .AddMvc()               // API version-aware extensions for MVC Core
        .AddApiExplorer()       // API version-aware API Explorer extensions
        .AddOData()             // API versioning extensions for OData
        .AddODataApiExplorer(); // API version-aware API Explorer extensions for OData

```

Each one also has the same or similar configuration options that you're used to. The default configuration for API Explorer support is:

```c#
services.AddApiVersioning().AddApiExplorer();
```

I hope that helps clear things up.

**Reply by andrei-micuda** [reply]
Yes, I eventually figured out that the extension methods are now chaining. I was only looking at the documentation and that is what confused me.

Thank you for the clarification!
